src/poetry/utils/password_manager.py

```diff
@@ -46,9 +46,14 @@ def get_credential(
             return default
 
         import keyring
+        import keyring.errors
 
         for name in names:
-            credential = keyring.get_credential(name, username)
+            try:
+                credential = keyring.get_credential(name, username)
+            except keyring.errors.KeyringLocked:
+                self._is_available = False
+                break
```

Comment:
This flow-control doesn't make sense (we shouldn't abort trying all `name`s) -- instead, defaulting credential to `None` and logging make more sense.

Reply:
If the keyring is locked because the user canceled, asking them again and again is not very good UX. A single `poetry install` command might prompt the user a hundred times...